### ISSUE TITLE
net: Failed to open nl80211 socket

### ISSUE THREAD

**Issue by TimB87** [greshake/i3status-rust#1742]
Heya,

congrats on the big release!

Converting the config was easy, but the net block seems to not work for me.
Error message: `Failed to open nl80211 socket`
![image](https://user-images.githubusercontent.com/7693498/220451190-c2c4bb92-4699-4058-8284-869d680cec18.png)

The binary was compiled with the --release flag, using `rustc 1.67.1 (d5a82bbd2 2023-02-07) (CRUX)` and pulseaudio features.

```
$ ip l
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: enp4s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq state UP mode DEFAULT group default qlen 1000
    link/ether 00:1b:21:19:09:ce brd ff:ff:ff:ff:ff:ff

$ ls /sys/class/net/
enp4s0  lo
```

In my config I have tried the following:
```
[[block]]
block = "net"
device = "enp4s0"
#[[block]]
#block = "net"
#device = "^enp4s0$"
#[[block.click]]
#button = "left"
#cmd = "/usr/bin/footclient -a floating_foot nmtui"
```

I just tried a debug binary with `RUST_LOG=net=debug i3status-rs` but it doesn't seem to output any more info as in why it fails.

Edit:
The debug info just went under for me in the flood of text :grin: 
```
$ i3status-rs | grep DEBUG
[2023-02-21T20:31:06Z DEBUG neli::socket] Message sent:
    Nlmsghdr { nl_len: 32, nl_type: Ctrl, nl_flags: NlmFFlags(FlagBuffer(5, PhantomData<neli::consts::nl::NlmF>)), nl_seq: 0, nl_pid: 0, nl_payload: Payload(Genlmsghdr { cmd: Getfamily, version: 2, reserved: 0, header: NoUserHeader, attrs: GenlBuffer([Nlattr { nla_len: 12, nla_type: AttrType { nla_nested: false, nla_network_order: false, nla_type: FamilyName }, nla_payload: Buffer }]) }) }
[2023-02-21T20:31:06Z DEBUG neli::socket] Message received: Nlmsghdr { nl_len: 52, nl_type: UnrecognizedConst(2), nl_flags: NlmFFlags(FlagBuffer(0, PhantomData<neli::consts::nl::NlmF>)), nl_seq: 0, nl_pid: 6765, nl_payload: Err(Nlmsgerr { error: -2, nlmsg: NlmsghdrErr { nl_len: 32, nl_type: Ctrl, nl_flags: NlmFFlags(FlagBuffer(5, PhantomData<neli::consts::nl::NlmF>)), nl_seq: 0, nl_pid: 0, nl_payload: Genlmsghdr { cmd: Getfamily, version: 2, reserved: 0, header: NoUserHeader, attrs: GenlBuffer([Nlattr { nla_len: 12, nla_type: AttrType { nla_nested: false, nla_network_order: false, nla_type: FamilyName }, nla_payload: Buffer }]) } } }) }
```

**Comment by MaxVerevkin** [comment]
What is the output of `iw dev`? By the way, `enp4s0` is a wired interface, right?

**Comment by TimB87** [comment]
enp4s0 is a wired interface, correct. I do not have `iw` installed.

**Comment by TimB87** [comment]
I just compiled a binary and maybe I'm missing something in my kernel? :grimacing: 

```
$ ./iw dev
nl80211 not found.
```

**Comment by TimB87** [comment]
I just checked, it might be `CFG80211` which I don't have on this machine as I do not have a wirelss device installed.
I can test this theory tomorrow.

```
 .config - Linux/x86 6.0.2 Kernel Configuration
 > Search (cfg80211) ──────────────────────────────────────────────────────────────────────────────────
  ┌──────────────────────────────────────── Search Results ─────────────────────────────────────────┐
  │ Symbol: CFG80211 [=n]                                                                           │
  │ Type  : tristate                                                                                │
  │ Defined at net/wireless/Kconfig:20                                                              │
  │   Prompt: cfg80211 - wireless configuration API                                                 │
  │   Depends on: NET [=y] && WIRELESS [=y] && (RFKILL [=m] || !RFKILL [=m])                        │
  │   Location:                                                                                     │
  │     Main menu                                                                                   │
  │       -> Networking support (NET [=y])                                                          │
  │ (1)     -> Wireless (WIRELESS [=y])                                                             │
  │ Selects: FW_LOADER [=y] && CRC32 [=y] && CRYPTO_SHA256 [=y]
```

**Comment by MaxVerevkin** [comment]
> I'm missing something in my kernel? 

`nl80211` is used to fetch wifi information. Since you don't use wireless net interfaces this is OK.

**Comment by MaxVerevkin** [comment]
Should be fixed by 0dbaf78f01e6c8f863f52806b2bcb65f6881a317

**Comment by TimB87** [comment]
Can confirm, that does fix it! Thanks a lot @MaxVerevkin
